### PR TITLE
Ensure nested field support

### PR DESCRIPTION
**Pull request by alallema**
Add tests on nested fields to ensure it is still maintained in Meilisearch.

### CODE REVIEW

File: index_search_test.go [hunk 1]
```diff
@@ -916,3 +916,172 @@ func TestIndex_SearchWithSort(t *testing.T) {
...
+					},
+				},
+				NbHits:           1,
+				Offset:           0,
+				Limit:            20,
+				ExhaustiveNbHits: false,
+			},
+		},
+		{
+			name: "TestIndexBasicSearchOnNestedFieldsWithCustomClient",
```
Comment (bidoubiwa) [inline index_search_test.go:956]: what is a custom client?

File: index_search_test.go [hunk 2]
```diff
@@ -916,3 +916,172 @@ func TestIndex_SearchWithSort(t *testing.T) {
...
+				searchableAttribute: []string{
+					"title", "info.comment",
+				},
+			},
+			want: &SearchResponse{
+				Hits: []interface{}{
+					map[string]interface{}{
+						"id": float64(5), "title": "The Hobbit",
+					},
+				},
```
Comment (bidoubiwa) [inline index_search_test.go:1014]: should you test the nested object as well?

File: main_test.go [hunk 3]
```diff
@@ -133,6 +133,33 @@ func SetUpBasicIndex(indexUID string) {
 	}
 }
 
+func SetUpIndexWithNestedFields(indexUID string) {
+	client := NewClient(ClientConfig{
+		Host:   "http://localhost:7700",
+		APIKey: masterKey,
```
Comment (bidoubiwa) [inline main_test.go:139]:
```suggestion
		Host: "http://localhost:7700",
		APIKey: masterKey,
```

**Review by bidoubiwa — CHANGES_REQUESTED** [review]

File: index_search_test.go (hunk 1 above)
Comment (alallema) [reply index_search_test.go:956]: We currently use the `fasthttp` library to create an `http` client but in go you have the native `net/http` library to create a client. We offer the possibility for the user to generate his own client with `net/http` or another library.

File: index_search_test.go (hunk 2 above)
Comment (alallema) [reply index_search_test.go]: Done !

**Pull request merged**